core/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java

```diff
@@ -247,18 +247,15 @@ protected XContentBuilder toInnerXContent(XContentBuilder builder, Params params
         List<GeoQueryContext> queryContexts = new ArrayList<>();
         Token token = parser.nextToken();
         if (token == Token.START_OBJECT || token == Token.VALUE_STRING) {
-            queryContexts.add(GeoQueryContext.parse(parser));
+            queryContexts.add(GeoQueryContext.PROTOTYPE.fromXContext(parser));
         } else if (token == Token.START_ARRAY) {
             while (parser.nextToken() != Token.END_ARRAY) {
-                queryContexts.add(GeoQueryContext.parse(parser));
+                queryContexts.add(GeoQueryContext.PROTOTYPE.fromXContext(parser));
             }
         }
         List<QueryContext> queryContextList = new ArrayList<>();
         for (GeoQueryContext queryContext : queryContexts) {
-            int minPrecision = this.precision;
-            if (queryContext.getPrecision() != -1) {
-                minPrecision = Math.min(minPrecision, queryContext.getPrecision());
-            }
```

Comment:
This is due to the change in the default `precision` for `GeoQueryContext.Builder` see [here](https://github.com/elastic/elasticsearch/pull/16398/files#diff-6252417c21bed452f272fe142129fed7R192). before `-1` indicated the index-time precision value be used for query contexts, now we set the default to `12` (highest precision) and take the minimum precision between the index and query time precision.